src/core/thread/thread_netif.cpp

```diff
@@ -171,6 +172,28 @@ ThreadError ThreadNetif::RouteLookup(const Ip6::Address &source, const Ip6::Addr
     return error;
 }
 
+ThreadError ThreadNetif::TmfFilter(const otMessage *aMessage, const otMessageInfo *aMessageInfo)
```

Comment:
I also wavered over the layer of this functionality before choosing the current implementation.  I strongly support the general "fire walling" feature. I've changed the way to pass both arguments.